pkg/asset/agent/manifests/agentclusterinstall.go

```diff
@@ -112,22 +145,49 @@ func (a *AgentClusterInstall) Generate(dependencies asset.Parents) error {
 			},
 		}
 
+		icOverridden := false
+		icOverrides := agentClusterInstallInstallConfigOverrides{}
 		if installConfig.Config.FIPS {
-			agentClusterInstall.SetAnnotations(map[string]string{
-				installConfigOverrides: `{ "fips": true }`,
-			})
+			icOverridden = true
+			icOverrides.FIPS = installConfig.Config.FIPS
 		}
 
-		setNetworkType(agentClusterInstall, installConfig.Config, "NetworkType is not specified in InstallConfig.")
+		if installConfig.Config.Platform.BareMetal != nil {
+			if len(installConfig.Config.Platform.BareMetal.APIVIPs) > 1 {
+				icOverridden = true
+				icOverrides.Platform = &agentClusterInstallPlatform{
+					BareMetal: &agentClusterInstallOnPremPlatform{
+						APIVIPs:     installConfig.Config.Platform.BareMetal.APIVIPs,
+						IngressVIPs: installConfig.Config.Platform.BareMetal.IngressVIPs,
+					},
+				}
+			}
+			agentClusterInstall.Spec.APIVIP = installConfig.Config.Platform.BareMetal.APIVIPs[0]
+			agentClusterInstall.Spec.IngressVIP = installConfig.Config.Platform.BareMetal.IngressVIPs[0]
+		} else if installConfig.Config.Platform.VSphere != nil {
+			if len(installConfig.Config.Platform.VSphere.APIVIPs) > 1 {
+				icOverridden = true
+				icOverrides.Platform = &agentClusterInstallPlatform{
+					VSphere: &agentClusterInstallOnPremPlatform{
+						APIVIPs:     installConfig.Config.Platform.VSphere.APIVIPs,
+						IngressVIPs: installConfig.Config.Platform.VSphere.IngressVIPs,
+					},
+				}
+			}
+			agentClusterInstall.Spec.APIVIP = installConfig.Config.Platform.VSphere.APIVIPs[0]
+			agentClusterInstall.Spec.IngressVIP = installConfig.Config.Platform.VSphere.IngressVIPs[0]
+		}
 
-		// TODO: Handle the case where both IPv4 and IPv6 VIPs are specified
-		apiVIP, ingressVIP := getVIPs(&installConfig.Config.Platform)
+		setNetworkType(agentClusterInstall, installConfig.Config, "NetworkType is not specified in InstallConfig.")
 
-		// set APIVIP and IngressVIP only for non SNO cluster for Baremetal and Vsphere platforms
-		// SNO cluster is determined by number of ControlPlaneAgents which should be 1
-		if int(*installConfig.Config.ControlPlane.Replicas) > 1 && apiVIP != "" && ingressVIP != "" {
-			agentClusterInstall.Spec.APIVIP = apiVIP
-			agentClusterInstall.Spec.IngressVIP = ingressVIP
+		if icOverridden {
+			overrides, err := json.Marshal(icOverrides)
+			if err != nil {
+				return errors.Wrap(err, "failed to marshal AgentClusterInstall installConfigOverrides")
+			}
+			agentClusterInstall.SetAnnotations(map[string]string{
+				installConfigOverrides: fmt.Sprintf("%s", overrides),
```

Comment:
gostatic check `the argument's underlying type is a slice of bytes, should use a simple conversion instead of fmt.Sprintf`

Nit: `installConfigOverrides: string(overrides)`